src/python/module/z5py/dataset.py

```diff
@@ -515,6 +515,44 @@ def find_maximum_coordinates(self, dim):
         """
         return self._impl.findMaximumCoordinates(dim)
 
+    def read_direct(self, dest, source_sel=None, dest_sel=None):
+        """ Wrapper to improve similarity to h5py. Reads from the dataset to ``dest``, using ``read_subarray``.
+
+        Args:
+            dest (array) destination object into which the read data is written to.
+            dest_sel (slice array) selection of data to write to ``dest``. Defaults to the whole range of ``dest``.
+            source_sel (slice array) selection in dataset to read from. Defaults to the whole range of the dataset.
+        Spaces, defined by ``source_sel`` and ``dest_sel`` must be in the same size but dont need to have the same
+        offset
+        """
+        rTsl1 = lambda x, y: slice(x.__int__(), y.__int__())
+        if source_sel is None:
+            source_sel = tuple(map(rTsl1, [0 for s in self.shape], self.shape))
+        if dest_sel is None:
+            dest_sel = tuple(map(rTsl1, [0 for s in dest.shape], dest.shape))
```

Comment:
I think I get what you are doing here, but this code seems way to complicated.
You could achieve the same with
``` python
if source_sel is None:
    source_sel = tuple(slice(0, sh) for sh in self.shape)
```
and similar for `dest_sel`.